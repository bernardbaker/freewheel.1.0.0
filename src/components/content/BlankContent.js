import React, {Component} from 'react'

class BlankContent extends Component {

  render() {
    return (
      <div>
        <h1>{this.props.message}</h1>
      </div>
    )
  }
}

export default BlankContent