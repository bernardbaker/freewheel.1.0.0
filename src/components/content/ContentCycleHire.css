.Input-Button {
  margin-left: 20px;
}

.Cycle-Search-Result {
  font-size: 0.625em;
}

ul,
li {
  margin-left: 0;
  list-style: none;
  padding-left: 0;
}

li {
  padding-top: 5px;
  padding-bottom: 5px;
  padding-left: 5px;
}

li:nth-child(odd) {
  background-color: rgb(230,230,230);
}

li:nth-child(even) {
  background-color: rgb(220,220,220);
}

.Result-List {
  height: 246px;
  display: block;
  overflow-y: auto;
  border-radius: 10px;
}