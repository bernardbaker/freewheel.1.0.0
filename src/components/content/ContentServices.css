.Messages {
  background-color: rgb(240,240,240);
  padding: 10px;
  overflow-y: auto;
  height: 230px;
  border-radius: 10px;
}