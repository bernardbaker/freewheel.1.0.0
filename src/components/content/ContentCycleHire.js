import React, {Component} from 'react'
import './ContentCycleHire.css'

class ContentCycleHire extends Component {

  constructor(props) {
    super(props)
    
    this.state = {
      results: [],
      searched: this.props.searched
    }
  }

  static getDerivedStateFromProps(nextProps, prevState){
    return Object.assign({}, prevState, {results: nextProps.results, searched: nextProps.searched})
  }

  render() {
    return (
      <div>
        <h1>Cycle hire content</h1>
        <input id="input" type="text" className="Cycle-Hire-Query-Input"/>
        <button className="Input-Button" onClick={this.props.queryCycleHireLocation}>Find cycles</button>
        <div>
            <p>
              {
                this.state.results.length ? this.state.results.length + ' search results' : this.state.searched === true ? 'Sorry no results for your search' : ''
              }
            </p>
            <ul className="Result-List">
            {
              this.state.results.map((element, index, array) => {
                return <li key={"result-"+index}><p className="Cycle-Search-Result">{element.id.substr(element.id.search('_')+1) + ' ' + element.commonName + ' (' + element.lat + ', ' + element.lon + ')'}</p></li>
              })
            }
            </ul>
        </div>
      </div>
    )
  }
}

export default ContentCycleHire