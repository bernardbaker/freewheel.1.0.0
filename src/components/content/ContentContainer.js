import React from 'react'
import ContentServices from './ContentServices'
import ContentCycleHire from './ContentCycleHire'

export default function (Component) {
  return class Container extends React.Component {

    constructor(props) {
      super(props)
      this.state = {
        content: this.props.reducers.service,
        queryCycleHireLocationResults: [],
        searched: false
      }

      this.queryCycleHireLocation = this.queryCycleHireLocation.bind(this)
    }

    queryCycleHireLocation(event) {
      if(document.getElementById('input').value.toString().length < 1) {
        this.setState(prevState => ({
          queryCycleHireLocationResults: prevState.queryCycleHireLocationResults = [],
          searched: prevState.searched = true
        }))
        return
      } 
      const query = document.getElementById('input').value
      fetch(`https://api.tfl.gov.uk/BikePoint/Search?query=${query}`)
      .then(response => {
        let res = response.json();
        res.then(payload => {
          this.setState(prevState => ({
            queryCycleHireLocationResults: prevState.queryCycleHireLocationResults = payload,
            searched: prevState.searched = true
          }))
        })
      })
    }

    static getDerivedStateFromProps(nextProps, prevState){
      return Object.assign({}, prevState, {content: nextProps.reducers.service})
    }
    
    render() {
      
      switch(this.state.content) {
        case 'cycle-hire' :
          return <ContentCycleHire queryCycleHireLocation={this.queryCycleHireLocation} results={this.state.queryCycleHireLocationResults} searched={this.state.searched}/>
        case 'service-update' :
          return <ContentServices {...this.props} />
        default :
          return <Component message="Welcome to the TFL info widget" />
      }
    }

    componentDidCatch(error, info) {
      console.log(error)
      console.log(info)
    }

    componentWillUnmount() {
      
    }
  }  
}