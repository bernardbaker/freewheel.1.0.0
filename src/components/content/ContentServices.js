import React, {Component} from 'react'
import './ContentServices.css'

class ContentServices extends Component {

  constructor(props) {
    super(props)
    
    this.state = {
      data: [],
      service: this.props.reducers.category,
      header: "",
      messages: []
    }
  }

  static getDerivedStateFromProps(nextProps, prevState){
    const data = nextProps.reducers.data.filter(e => e.id === nextProps.reducers.category)
    const header = data[0].lineStatuses.filter(e => e.statusSeverity !== 10).length === 0 ? "No service disruptions" : "Service currently suffering disruptions"
    let messages = data[0].lineStatuses.filter(e => e.statusSeverity !== 10)
    return Object.assign({}, prevState, {header, messages})
  }

  render() {
    return (
      <div>
        <h1>Service update</h1>
        <h3>{this.state.header}</h3>
        <div className={this.state.messages.length > 0 ? "Messages" : ''}>
        {
          this.state.messages.map((element, index, array) => {
            return <div key={"reason-"+index}><p>{element.reason}</p><br/></div>
          })
        }
        </div>
      </div>
    )
  }
}

export default ContentServices