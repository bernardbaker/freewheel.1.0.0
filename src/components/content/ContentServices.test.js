import React from 'react';
import ReactDOM from 'react-dom';
import ContentContainer from './ContentContainer';
import BlankContent from './BlankContent'
import ContentServices from './ContentServices'
import ContentCycleHire from './ContentCycleHire'

import { shallow } from 'enzyme';
import renderer from 'react-test-renderer'
import data from '../../mock/data.json'

describe('ContentContainer component using content services for presentational component', () => {
  it('matches the snapshot', () => {
    const Component = ContentContainer(ContentServices)
    const props = {reducers:{category:'dlr', service:'service-update', data:data}}
    const tree = renderer.create(<Component {...props}/>).toJSON()
    expect(tree).toMatchSnapshot()
  })
  
  it('default data when content services presentational component is created', () => {
    const Component = ContentContainer(ContentServices)
    const props = {reducers:{category:'dlr', service:'service-update', data:data}}
    const wrapper = shallow(<Component {...props}/>)
    expect(wrapper.state().content).toEqual('service-update')
  })  
})

describe('ContentContainer component using blank services for presentational component', () => {
  it('matches the snapshot', () => {
    const Component = ContentContainer(BlankContent)
    const props = {reducers:{category:'dlr', service:null, data:data}}
    const tree = renderer.create(<Component {...props} />).toJSON()
    expect(tree).toMatchSnapshot()
  })
  
  it('default data when content services presentational component is created', () => {
    const Component = ContentContainer(BlankContent)
    const props = {reducers:{category:'dlr', service:null, data:data}}
    const wrapper = shallow(<Component {...props} />)
    expect(wrapper.state().content).toEqual(null)
  })  
})

describe('ContentContainer component using blank services for presentational component', () => {
  it('matches the snapshot', () => {
    const Component = ContentContainer(ContentCycleHire)
    const props = {reducers:{category:'cycle-hire', service:'cycle-hire', data:data}}
    const tree = renderer.create(<Component {...props} />).toJSON()
    expect(tree).toMatchSnapshot()
  })
  
  it('default data when content services presentational component is created', () => {
    const Component = ContentContainer(ContentCycleHire)
    const props = {reducers:{category:'cycle-hire', service:'cycle-hire', data:data}}
    const wrapper = shallow(<Component {...props} />)
    expect(wrapper.state().content).toEqual('cycle-hire')
  })  
})