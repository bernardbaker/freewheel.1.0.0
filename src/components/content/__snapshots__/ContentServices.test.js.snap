// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ContentContainer component using blank services for presentational component matches the snapshot 1`] = `
<div>
  <h1>
    Welcome to the TFL info widget
  </h1>
</div>
`;

exports[`ContentContainer component using blank services for presentational component matches the snapshot 2`] = `
<div>
  <h1>
    Cycle hire content
  </h1>
  <input
    className="Cycle-Hire-Query-Input"
    id="input"
    type="text"
  />
  <button
    className="Input-Button"
    onClick={[Function]}
  >
    Find cycles
  </button>
  <div>
    <p>
      
    </p>
    <ul
      className="Result-List"
    />
  </div>
</div>
`;

exports[`ContentContainer component using content services for presentational component matches the snapshot 1`] = `
<div>
  <h1>
    Service update
  </h1>
  <h3>
    Service currently suffering disruptions
  </h3>
  <div
    className="Messages"
  >
    <div>
      <p>
        Docklands Light Railway: Train service will resume later this morning. 
      </p>
      <br />
    </div>
  </div>
</div>
`;
